src/gcinfo/gcinfoencoder.cpp

```diff
@@ -1787,34 +1842,16 @@ void GcInfoEncoder::Build()
                     *newLiveState = liveState;
                     hashMap.Set(newLiveState, (UINT32)(-1));
                 }
-
-
-                if(++callSiteIndex == m_NumCallSites)
-                    break;
-            
-                callSite = m_pCallSites[callSiteIndex];
-            }
-            else
-            {
-                UINT32 slotIndex = pCurrent->SlotId;
-                BYTE becomesLive = pCurrent->BecomesLive;
-                _ASSERTE((liveState.ReadBit(slotIndex) && !becomesLive)
-                        || (!liveState.ReadBit(slotIndex) && becomesLive));
-                liveState.WriteBit(slotIndex, becomesLive);
-                pCurrent++;
             }
         }
+        catch (GcInfoNoMemoryException& e)
+        {
+            outOfMemory = true;
+        }
 
-        // Check for call sites at offsets past the last transition
-        if (callSiteIndex < m_NumCallSites)
+        if (outOfMemory)
         {
-            UINT32 liveStateOffset = 0;
-            if (!hashMap.Lookup(&liveState, &liveStateOffset))
-            {
-                BitArray * newLiveState = new (m_pAllocator->Alloc(sizeof(BitArray))) BitArray(m_pAllocator, size_tCount);
-                *newLiveState = liveState;
-                hashMap.Set(newLiveState, (UINT32)(-1));
-            }
+            m_pNoMem();
         }
 
```

Comment:
Yes, but that change is quite a bit more invasive: every user of the GC info encoder would need to be updated to supply the function as e.g. a type trait. The proposed change does not require this.